baseplate/experiments/providers/__init__.py

```diff
@@ -119,6 +127,35 @@ def parse_experiment(config):
             version=version,
             config=experiment_config,
         )
+    elif experiment_type == "multi_variant":
+        return MultiVariantExperiment.from_dict(
```

Comment:
I'd probably assign the class to something like experiment_type and then return experiment_type.from_dict(...) to DRY this up